### DISCUSSION TITLE
How to set/reset the defaultValue to an <input type="date" /> ?

### DISCUSSION THREAD

**Discussion by Humadroid** [react-hook-form/react-hook-form#5595] [Q&A] [answered]
How to set/reset the defaultValue to an `<input type="date" />` ?
[**Stackblitz Link to issue**](https://stackblitz.com/edit/react-hook-form-input-type-date-defaultvalue-issue?file=src/App.js)

**Comment by bluebill1049** [comment, marked as answer]
have you tried https://react-hook-form.com/api/useform/reset api?

**Reply by Humadroid** [reply]
```
const defaultValues = { dob: new Date('07/21/1992') };

// Usage with Controller

          <Controller
            name="dob"
            control={control}
            rules={{
              required: {
                value: true,
                message: 'Enter Date of birth'
              }
            }}
            render={({ field: { ref, ...rest } }) => (
              <input type="date" {...rest} />
            )}
          />
         <button type="reset" onClick={() => reset({ ...defaultValues })}>
            Reset
         </button>
```
This is how I've done in the [Stackblitz link](https://stackblitz.com/edit/react-hook-form-input-type-date-defaultvalue-issue?file=src/App.js) above ..this is also like the  link to [reset API](https://react-hook-form.com/api/useform/reset) 
Is this not the right approach @bluebill1049 ?

**Reply by bluebill1049** [reply]
https://codesandbox.io/s/dank-http-74hk4?file=/src/App.js ?

**Reply by Humadroid** [reply]
Yes, that works with register, and even with Controller, _I just copied over my code from stackblitz to codesandbox and it works_, ~not sure what I was missing!~, Okay I noted the differences and the above stackblitz link works too !
[**Codesandbox working link**](https://codesandbox.io/s/stoic-joliot-hvb5i?file=/src/App.js)
~Not sure what it was but anyway~ thanks @bluebill1049 for the help 👍🏼💯
